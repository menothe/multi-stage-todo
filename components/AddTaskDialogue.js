import React from 'react';
import TextField from 'material-ui/TextField';
import Dialog from 'material-ui/Dialog';
import RaisedButton from 'material-ui/RaisedButton';


// popup for user to submit a new task
export const AddTaskDialogue = ({ visible, addTask }) => (
  <Dialog modal={true} open={visible}
    actions={[
      <RaisedButton onClick={addTask} label='Add Task' primary={true} style={{ float: 'right', marginBottom: 15 }} />
    ]}
    >
    <TextField id="newTaskName" fullWidth={true} hintText={"Task Name"} />
    <br />
    <TextField id="newTaskDescription" fullWidth={true} hintText={"Task Description"} />
  </Dialog>
);
