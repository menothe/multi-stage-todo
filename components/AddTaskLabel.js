import React from 'react';
import FloatingActionButton from 'material-ui/FloatingActionButton';
import ContentAdd from 'material-ui/svg-icons/content/add';


// label on left hand of screen to trigger <AddTaskDialogue />
export const AddTaskLabel = ({ showAddTaskDialogue }) => (
  <div className={'column addTask'}>
    <h3>Add Task</h3>
    <FloatingActionButton onClick={showAddTaskDialogue} mini={true} style={{ marginLeft: 10 }}>
      <ContentAdd />
    </FloatingActionButton>
  </div>
)
