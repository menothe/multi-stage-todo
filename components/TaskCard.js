import React from 'react';
import { Card, CardActions, CardHeader, CardText } from 'material-ui/Card';
import RaisedButton from 'material-ui/RaisedButton';
import { typeActionLookup } from '../actions/actions';


// material-ui card rendering details about task
export const TaskCard = ({ id, state, name, description, progressTask }) => (
  <Card className={'card'}>
    <CardHeader title={name} subtitle={id} />
    <CardText>{description}</CardText>
    <CardActions>
      <RaisedButton onClick={() => progressTask(id)} label={typeActionLookup[state]} primary={true} style={{ float: 'right', marginBottom: 15 }} />
    </CardActions>
  </Card>
);
