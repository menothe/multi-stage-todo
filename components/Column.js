import React from 'react';
import { TaskCard } from './TaskCard';

export const Column = ({ label, tasks, state, progressTask }) => (
  <div className={'column'}>
    <h3>{label}</h3>
    {tasks && tasks.filter(t => t.state === state).map((t, i) => (
      <TaskCard state={t.state} progressTask={progressTask} key={i} name={t.taskName} id={t.id} description={t.taskDescription} />
    ))}
  </div>
);
