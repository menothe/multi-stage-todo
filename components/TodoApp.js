import React, { Component } from 'react';
import { AddTaskLabel } from './AddTaskLabel';
import { TaskCard } from './TaskCard';
import { AddTaskDialogue } from './AddTaskDialogue';
import { Column } from './Column';
import { nextTypeLookup } from '../actions/actions';


// main App Class
export default class TodoApp extends Component {

  state = {
    tasks: [],
    addTaskDialogueVisible: false
  }

  showAddTaskDialogue = () => {
    this.setState(() => ({ addTaskDialogueVisible: true }));
  }

  hideAddTaskDialogue = () => {
    this.setState(() => ({ addTaskDialogueVisible: false }));
  }

  // creates new task based on text entered in <AddTaskDialogue />
  addTaskToState = () => {
    let { tasks } = this.state;
    let newTask = {
      taskName: document.getElementById('newTaskName').value,
      taskDescription: document.getElementById('newTaskDescription').value,
      id: tasks.length + 1,
      state: 'todo'
    };

    tasks[tasks.length] = newTask;
    this.setState(() => ({ tasks, addTaskDialogueVisible: false }));
  }

  // updates state of task based on current state
  progressTask = (tId) => {
    let { tasks } = this.state;
    for (let i = 0; i < tasks.length; i++) {
      if (tasks[i].id === tId) {
        tasks[i].state = nextTypeLookup[tasks[i].state];
        break;
      }
    }
    this.setState(() => ({ tasks }));
  }

  // seeds state w/ initial app data
  componentDidMount() {
    fetch('https://private-caaa7-versustracker.apiary-mock.com/tasks')
      .then((res) => res.json())
      .then((items) => this.setState(() => ({ tasks: items.tasks })));
  }

  render() {
    let { tasks, addTaskDialogueVisible } = this.state;

    return (
      <div className={'container'}>
        <AddTaskLabel showAddTaskDialogue={this.showAddTaskDialogue}/>
        <AddTaskDialogue visible={addTaskDialogueVisible} addTask={this.addTaskToState} hideDialog={this.hideAddTaskDialogue}/>
        <Column label='todo' tasks={tasks} state={'todo'} progressTask={this.progressTask}/>
        <Column label='in-progress' tasks={tasks} state={'in-progress'} progressTask={this.progressTask}/>
        <Column label='complete' tasks={tasks} state={'complete'} progressTask={this.progressTask}/>
      </div>
    )
  }
}
