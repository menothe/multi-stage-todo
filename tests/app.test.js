import React from 'react';
import { AddTaskLabel } from '../components/AddTaskLabel';
import { AddTaskDialogue } from '../components/AddTaskDialogue';
import { Column } from '../components/Column';
import { TaskCard } from '../components/TaskCard';
import ReactShallowRenderer from 'react-test-renderer/shallow';

test('should render AddTaskLabel correctly', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<AddTaskLabel />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('should render AddTaskDialogue correctly', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<AddTaskDialogue visible={undefined}/>);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('should render Column correctly', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<Column />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('should render TaskCard correctly', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<TaskCard />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
