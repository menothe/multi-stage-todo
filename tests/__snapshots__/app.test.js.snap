// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddTaskDialogue correctly 1`] = `
<Dialog
  actions={
    Array [
      <RaisedButton
        disabled={false}
        fullWidth={false}
        label="Add Task"
        labelPosition="after"
        onClick={undefined}
        primary={true}
        secondary={false}
        style={
          Object {
            "float": "right",
            "marginBottom": 15,
          }
        }
      />,
    ]
  }
  autoDetectWindowHeight={true}
  autoScrollBodyContent={false}
  modal={true}
  open={undefined}
  repositionOnUpdate={true}
>
  <TextField
    disabled={false}
    floatingLabelFixed={false}
    fullWidth={true}
    hintText="Task Name"
    id="newTaskName"
    multiLine={false}
    rows={1}
    type="text"
    underlineShow={true}
  />
  <br />
  <TextField
    disabled={false}
    floatingLabelFixed={false}
    fullWidth={true}
    hintText="Task Description"
    id="newTaskDescription"
    multiLine={false}
    rows={1}
    type="text"
    underlineShow={true}
  />
</Dialog>
`;

exports[`should render AddTaskLabel correctly 1`] = `
<div
  className="column addTask"
>
  <h3>
    Add Task
  </h3>
  <FloatingActionButton
    disabled={false}
    mini={true}
    onClick={undefined}
    secondary={false}
    style={
      Object {
        "marginLeft": 10,
      }
    }
    zDepth={2}
  >
    <ContentAdd />
  </FloatingActionButton>
</div>
`;

exports[`should render Column correctly 1`] = `
<div
  className="column"
>
  <h3 />
</div>
`;

exports[`should render TaskCard correctly 1`] = `
<Card
  className="card"
  expandable={false}
  expanded={null}
  initiallyExpanded={false}
>
  <CardHeader
    avatar={null}
    subtitle={undefined}
    title={undefined}
  />
  <CardText />
  <CardActions>
    <RaisedButton
      disabled={false}
      fullWidth={false}
      label={undefined}
      labelPosition="after"
      onClick={[Function]}
      primary={true}
      secondary={false}
      style={
        Object {
          "float": "right",
          "marginBottom": 15,
        }
      }
    />
  </CardActions>
</Card>
`;
